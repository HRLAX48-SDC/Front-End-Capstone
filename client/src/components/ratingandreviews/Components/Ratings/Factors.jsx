import React, { useState, useEffect} from 'react';
import Card from 'react-bootstrap/Card'


function Factors (props) {

    return (
      <Card>
        <Card.Title>
    
        </Card.Title>
        <Card.Body>
           <p>Size</p>
           <p>Too small--Perfect--Too large</p>
           <p>comfort</p>
           <p>Poor----Perfect</p>
        </Card.Body>
      </Card>
    )

}
export default Factors;