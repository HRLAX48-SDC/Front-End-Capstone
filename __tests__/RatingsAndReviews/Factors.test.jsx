import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Factors from '../../client/src/components/ratingandreviews/Components/Ratings/Factors.jsx'

describe('<Factors />', () =>{

    it('should correctly render factors', () => {
        const metaData = {
            characteristics :{
                Comfort: {id: 142034, value: "3.1627906976744186"},
                Fit: {id: 142032, value: "2.9021739130434783"},
                Length: {id: 142033, value: "2.8045977011494253"},
                Quality: {id: 142035, value: "3.1976744186046512"}
            },
            product_id: "42366"
        }
        
        const factors = renderer
        .create(<Factors  metaData={metaData} />)
        .toJSON();
        expect(factors).toMatchSnapshot();
    });
});


