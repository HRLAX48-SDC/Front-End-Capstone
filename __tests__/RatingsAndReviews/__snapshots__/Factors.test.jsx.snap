// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Factors /> should correctly render factors 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div>
      <div>
        <b>
          Comfort
        </b>
        <input
          className="form-range"
          defaultValue={63.25581395348837}
          disabled={true}
          type="range"
        />
        <div
          className="factors-text hstack gap-3"
        >
          <div>
            Uncomfortable
          </div>
          <div
            className="mx-auto"
          >
            Ok
          </div>
          <div
            className="ms-auto"
          >
            Perfect
          </div>
        </div>
      </div>
      <div>
        <b>
          Fit
        </b>
        <input
          className="form-range"
          defaultValue={58.04347826086958}
          disabled={true}
          type="range"
        />
        <div
          className="factors-text hstack gap-3"
        >
          <div>
            Runs tight
          </div>
          <div
            className="mx-auto"
          >
            Perfect
          </div>
          <div
            className="ms-auto"
          >
            Runs long
          </div>
        </div>
      </div>
      <div>
        <b>
          Length
        </b>
        <input
          className="form-range"
          defaultValue={56.09195402298851}
          disabled={true}
          type="range"
        />
        <div
          className="factors-text hstack gap-3"
        >
          <div>
            Runs short
          </div>
          <div
            className="mx-auto"
          >
            Perfect
          </div>
          <div
            className="ms-auto"
          >
            Runs long
          </div>
        </div>
      </div>
      <div>
        <b>
          Quality
        </b>
        <input
          className="form-range"
          defaultValue={63.95348837209303}
          disabled={true}
          type="range"
        />
        <div
          className="factors-text hstack gap-3"
        >
          <div>
            Poor
          </div>
          <div
            className="mx-auto"
          >
            What I Expected
          </div>
          <div
            className="ms-auto"
          >
            Perfect
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
